crates/aptos-rosetta/src/account.rs

```diff
@@ -138,7 +140,10 @@ async fn get_balances(
     }
 
     // Retrieve all account resources
-    // TODO: This will need to change for FungibleAssets, will need to lookup on a list of known FAs
+
+    // This is now less performant, but more flexible
+    // Retrieve stake pool and account sequence number
```

Comment:
Can we change it to read the two resources directly instead of retrieving everything and do a match?

Reply:
Sure, though the accounts should have so few resources, that making two calls may be less performant than one.

Reply:
That's probably not clear to the average user.  So, if that's the case with sharding, you may want to look at how to ensure the performance is similar or at least decent, since it will force users to change their flows.

Reply:
It will be decent if your query them separately. If you really care about the performance you can do it in multiple threads, but I don't think the user can tell the difference. In general we expect the call to read a resource directly takes less than 1ms.